### PR TITLE
fix(rest): not sending Content-Length header when the body is undefined

### PR DESCRIPTION
**Pull request by Helloyunho**
## About

This PR fixes the rest doesn't send the `Content-Length` header when the body is undefined or null.
Very related to https://github.com/denoland/deno/issues/11920

Also, I removed all status checking codes since harmony will throw an error if the status is not 2xx.

## Status

- [x] These changes have been tested against Discord API or do not contain API change.
- [ ] This PR includes only documentation changes, no code change.
- [ ] This PR introduces some Breaking changes.

### CODE REVIEW

File: src/structures/member.ts [hunk 1]
```diff
@@ -183,18 +183,10 @@ export class Member extends SnowflakeBase {
...
-      undefined,
-      null,
-      true,
-      {
-        reason
-      }
-    )
-    if (resp.ok !== true) return false
-    else return true
+    await this.client.rest.delete(GUILD_MEMBER(this.guild.id, this.id), {
```
Comment (DjDeveloperr) [inline src/structures/member.ts:186]: Sending body with DELETE :harold:

**Review by DjDeveloperr — CHANGES_REQUESTED** [review]

File: src/structures/member.ts (hunk 1 above)
Comment (Helloyunho) [reply src/structures/member.ts]: fixed

**Review by DjDeveloperr — APPROVED** [review]
LGTM

File: src/structures/guild.ts [hunk 2]
```diff
@@ -133,16 +133,11 @@ export class GuildBans extends Base {
    * @param user User to unban, ID or User object.
    */
   async remove(user: string | User): Promise<boolean> {
```
Comment (DjDeveloperr) [inline src/structures/guild.ts:135]: Supports audit reason btw https://discord.com/developers/docs/resources/guild#remove-guild-ban

Comment (Helloyunho) [reply src/structures/guild.ts:135]: kay imma add it too

**Pull request merged**